### PR TITLE
ci: run linux arm tests on CircleCI

### PR DESCRIPTION
**Pull request by jkleinsc**
#### Description of Change

This PR moves our arm/arm64 Linux testing to CircleCI.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by nornagon** [comment]
I'm confused why `base::internal::PCScanInternal::PCScanInternal` is being invoked at all, since `PA_ALLOW_PCSCAN` should be false on x86? https://source.chromium.org/chromium/chromium/src/+/main:base/allocator/partition_allocator/partition_alloc_config.h;l=25-27;drc=6efee78914660d0249e01aa403579476d8dc6a20

**Comment by jkleinsc** [comment]
@nornagon its because this call doesn't use that flag:
https://source.chromium.org/chromium/chromium/src/+/main:base/threading/platform_thread_posix.cc;drc=37a0e6c0a6a13a078d08f51faf276c4eb4f0ef1a;l=77

Maybe it should?

**Comment by nornagon** [comment]
Hm, yeah, it seems like it ought to 🤔  perhaps take a stab at upstreaming that & see what the reviewer has to say?

File: patches/chromium/don_t_run_pcscan_notifythreadcreated_if_pcscan_is_disabled.patch [hunk 1]
```diff
@@ -35,3 +55,33 @@ index dbf96f68f9d8ca74616af10c236ebe24984b2b5e..e11237ad35743e9cc724f4974662a5ed
...
+ #include "base/allocator/partition_allocator/starscan/pcscan.h"
+ #include "base/allocator/partition_allocator/starscan/stack/stack.h"
+ #endif
+@@ -104,7 +105,7 @@ DWORD __stdcall ThreadFunc(void* params) {
+                                 FALSE,
+                                 DUPLICATE_SAME_ACCESS);
+ 
+-#if BUILDFLAG(USE_PARTITION_ALLOC_AS_MALLOC)
++#if BUILDFLAG(USE_PARTITION_ALLOC_AS_MALLOC) && defined(PA_ALLOW_PCSCAN)
+   internal::PCScan::NotifyThreadCreated(internal::GetStackPointer());
```
Comment (nornagon) [inline patches/chromium/don_t_run_pcscan_notifythreadcreated_if_pcscan_is_disabled.patch:76]: Should this perhaps also check the feature `kPartitionAllocPCScan`?

**Comment by nornagon** [comment]
Looks like that was successful 🎉

File: patches/chromium/don_t_run_pcscan_notifythreadcreated_if_pcscan_is_disabled.patch (hunk 1 above)
Comment (jkleinsc) [reply patches/chromium/don_t_run_pcscan_notifythreadcreated_if_pcscan_is_disabled.patch:76]: Hmm not sure... I upstreamed the patch here:
https://chromium-review.googlesource.com/c/chromium/src/+/2965109

**Review by nornagon — APPROVED** [review]

**Comment by jkleinsc** [comment]
Merging as CI failure is unrelated to PR change.

**Pull request merged**